File: Python/interleave.py
def interleave(arr1, arr2):
  i = 0
  interwoven = []
  while i < len(arr1):
    interwoven.append(arr1[i])
    interwoven.append(arr2[i])
    i+=1

  print(interwoven)

if __name__ == "__main__":
    interleave(["a", "b", "c"], ["d", "e", "f"])

File: Go/interleave.go
package main

// This algorithm will assume that both arrays being passed in are the same length
func interleave(arr1 []string, arr2 []string) []string {
	interweave := make([]string, 0)

	i := 0
	for i < len(arr1) {
		interweave = append(interweave[:], arr1[i], arr2[i])
		i++
	}

	return interweave
}
